rust/src/cliwrap/rpm.rs

```diff
@@ -0,0 +1,64 @@
+use clap::{App, Arg};
+use failure::Fallible;
+
+use crate::cliwrap::cliutil;
+
+/// Set of RPM arguments that we know are safe to pass through,
+/// or that we handle internally.
+fn new_rpm_app<'r>() -> App<'r, 'static> {
+    let name = "cli-ostree-wrapper-rpm";
+    App::new(name)
+        .bin_name(name)
+        .version("0.1")
+        .about("Wrapper for rpm")
+        .arg(Arg::with_name("all").short("a"))
+        .arg(Arg::with_name("file").short("f"))
+        .arg(Arg::with_name("package").short("p"))
+        .arg(Arg::with_name("query").short("q"))
+        .arg(Arg::with_name("verify").short("V"))
+        .arg(Arg::with_name("version"))
+}
+
+/// Primary entrypoint to running our wrapped `rpm` handling.
+pub(crate) fn main(argv: &[&str]) -> Fallible<()> {
+    if cliutil::is_unlocked()? {
+        // For now if we're unlocked, just directly exec rpm. In the future we
+        // may choose to take over installing a package live.
+        cliutil::exec_real_wrapped("rpm", argv)
+    } else {
+        let mut app = new_rpm_app();
+        let mut with_warning = true;
+        if let Ok(matches) = app.get_matches_from_safe_borrow(argv) {
```

Comment:
Ahh OK, I think I understand the logic here.

But IIUC, this means that any query more complex than just `-qf` or `-qa` will trigger the warning & sleep, right? (E.g. `--requires`, `--whatprovides`, `--queryformat`, etc...). That's going to be frustrating. Let's tweak the logic here so that as long as `-q/--query` is provided, regardless of any other arg, we just pass through without warning? Not sure if `clap` can do this.

(Looking through `rpm(8)`.) Maybe also `--querytags` and `--showrc` which are two other obvious "read-only" operations.

Reply:
Hmm.  I looked at this; I think we can just parse the arguments twice and see whether `-q/--query` is present.  However, it does end up in an interesting corner case because:

```
walters@toolbox ~/s/d/r/ostree> rpm -e -i bash -qa
bash-5.0.7-3.fc31.x86_64
```

Apparently today `--query` overrides the presence of `-e/-i`, instead of e.g. erroring out as I'd expect.  Maybe we can just rely on that but...